### PR TITLE
[DO - 1720] Create PEP03 scoring rule

### PR DESCRIPTION
**Pull request by Andrii17s**
0)Create BaseScoringRule class
1)Create PEP03 scoring rule

### CODE REVIEW

File: business_register/pep_scoring/rules.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    def calculate_weight(self):
+        family_ids = self.pep.related_persons.filter(
+            to_person_links__category=RelatedPersonsLink.FAMILY,
+        ).values_list('id', flat=True).all()[::1]
+        family_ids.append(self.pep.id)
+        have_weight = Property.objects.filter(
+            declaration__pep_id__in=family_ids,
+            valuation__isnull=True,
+            type=Property.SUMMER_HOUSE,
+            created_at__year__gte=2015,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:40]: created_at is the time of storing the record in our DB. We should use PropertyRight.acquisition date field for this check

File: business_register/pep_scoring/rules.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    def calculate_weight(self):
+        family_ids = self.pep.related_persons.filter(
+            to_person_links__category=RelatedPersonsLink.FAMILY,
+        ).values_list('id', flat=True).all()[::1]
+        family_ids.append(self.pep.id)
+        have_weight = Property.objects.filter(
+            declaration__pep_id__in=family_ids,
+            valuation__isnull=True,
+            type=Property.LAND,
+            created_at__year__gte=2015,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:69]: same here

File: business_register/pep_scoring/rules.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    """
+
+    def calculate_weight(self):
+        family_ids = self.pep.related_persons.filter(
+            to_person_links__category=RelatedPersonsLink.FAMILY,
+        ).values_list('id', flat=True).all()[::1]
+        family_ids.append(self.pep.id)
+        have_weight = Vehicle.objects.filter(
+            declaration__pep_id__in=family_ids,
+            created_at__year__gte=2015,
```
Comment (IegorSoboliev) [inline business_register/pep_scoring/rules.py:96]: and here

Comment (IegorSoboliev) [reply business_register/pep_scoring/rules.py:96]: VehicleRight. acquisition_date

Comment (IegorSoboliev) [reply business_register/pep_scoring/rules.py:96]: VehicleRight. acquisition_date

**Review by IegorSoboliev — CHANGES_REQUESTED** [review]

File: business_register/pep_scoring/rules.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        have_weight = PropertyRight.objects.filter(
+            pep_id__in=family_ids,
+            property__valuation__isnull=True,
+            type=Property.SUMMER_HOUSE,
+            acquisition_date__year__gte=2015,
+        ).values_list('property_id', 'property__declaration_id').all()[::1]
+
+        if have_weight:
+            value = {
+                "weight": 0.4,
```
Comment (roman-tiukh) [inline business_register/pep_scoring/rules.py:46]: weight should be saved at another field not in json

File: business_register/pep_scoring/rules.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    Rule 3.1 - PEP03_home
+    weight - 0.4
+    There is no information on the value of the real estate owned by PEP or
+    family members since 2015
+    """
+
+    def calculate_weight(self):
+        family_ids = self.pep.related_persons.filter(
+            to_person_links__category=RelatedPersonsLink.FAMILY,
+        ).values_list('id', flat=True).all()[::1]
```
Comment (roman-tiukh) [inline business_register/pep_scoring/rules.py:35]: values_list('id', flat=True) already return list
```suggestion
        family_ids = self.pep.related_persons.filter(
            to_person_links__category=RelatedPersonsLink.FAMILY,
        ).values_list('id', flat=True)
```

**Review by IegorSoboliev — APPROVED** [review]

File: business_register/pep_scoring/rules.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            acquisition_date__year__gte=2015,
+        ).values_list('property_id', 'property__declaration_id')[::1]
+        if have_weight:
+            weight = 0.1
+            value = {
+                "property_id": have_weight[0][0],
+                "declaration_id": have_weight[0][1],
+            }
+            return json.dumps(value), weight
+        return 0
```
Comment (roman-tiukh) [inline business_register/pep_scoring/rules.py:80]: 1. no need to call `json.dumps`
2. function should return same result if don't `have_weight` 
like this ->
```suggestion
            return weight, data
        return 0, {}
```

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**